test/testdata/lsp/hover.rb

```diff
@@ -87,13 +87,6 @@ def tests_return_markdown
     # ^^^^^^^^^^^^^^^^^^^^^ hover-line: 6 ---
     # ^^^^^^^^^^^^^^^^^^^^^ hover-line: 8 Tests return markdown output
   end
-
-  sig {returns(T.attached_class)}
-  def self.factory
-    #      ^ hover: sig {returns(T.attached_class)}
-    #      ^ hover: def self.factory; end
-    new
-  end
```

Comment:
Should we keep this test here, but change the `hover:` assertion to show the correct behavior? Maybe you deleted it because that's already tested somewhere else?

Reply:
[The commit in which this was removed](https://github.com/sorbet/sorbet/pull/7048/commits/c916874ffd3e23f8cabe2ea4950de94fa09054e3) actually restores the hover behaviour that is currently on `master`; I had mistakenly changed the options that hover was using to show the method in an [earlier commit](https://github.com/sorbet/sorbet/pull/7048/commits/0849d83d4d1f8c316df051fdd32e4392691a1fa4) (specifically [this line](https://github.com/sorbet/sorbet/blob/0849d83d4d1f8c316df051fdd32e4392691a1fa4/main/lsp/requests/hover.cc#L71) incorrectly added `withShowForRBI()`[^1]).

But perhaps you're suggesting we actually change how hover works?

[^1]: Although `withShowForRBI()` was since renamed to `useValidSyntax` ced85cc1f.

Reply:
Oh, sorry, I think understand what you mean now: keep the addition I made to the test and ensure it has the correct hover assertion. Will fix!

Reply:
Fixed in 79fe4ac. Note that this asserts that invalid syntax is acceptable on hover, which I believe is what we want.